### PR TITLE
fix: resetting the insight result on cancel save(#16370)

### PR DESCRIPTION
**Pull request by Arun-chaitanya**
## Problem

Saved insight in bad state after navigation #16370

## Changes

Resetting the result on cancel the edit.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by thmsobrmlr** [comment]
@Arun-chaitanya Thanks for your work here! I've just started to review the PR, but can't reproduce the original issue on master or on your branch minus the commit with the fix any more. Were you still able to reproduce the issue?

**Comment by Arun-chaitanya** [comment]
@thmsobrmlr , there is actually 1 step missing "How to reproduce" section in https://github.com/PostHog/posthog/issues/16370
1. Follow the following steps
2. Create an insight with following configuration
(code mentioned in the original issue)
3. Reload the page
4. Click on "Edit" then "Funnels"
5. Now again click on "Trends" tab
6. Click on "Insights" in the breadcrumb navigation and confirm navigating away with unsaved changes
7. Navigate to the saved insight -> see that the result is NaN instead of a number

File: frontend/src/scenes/insights/insightDataLogic.ts [hunk 1]
```diff
@@ -189,7 +189,11 @@ export const insightDataLogic = kea<insightDataLogicType>([
         },
         cancelChanges: () => {
             const savedFilters = values.savedInsight.filters
+            const savedResult = values.savedInsight.result
             actions.setQuery(savedFilters ? queryFromFilters(savedFilters) : null)
+            if (savedResult) {
+                actions.setInsightData({ ...values.insightData, result: values.savedInsight.result })
+            }
```
Comment (thmsobrmlr) [inline frontend/src/scenes/insights/insightDataLogic.ts:196]:
```suggestion
actions.setInsightData({ ...values.insightData, result: savedResult ? savedResult : null })
```
We should probably reset the result in any case (wether we have data or not). Also reusing the `savedResult` var.

**Comment by Arun-chaitanya** [comment]
@thmsobrmlr please find the video 

https://github.com/PostHog/posthog/assets/44457256/52d88eca-9990-4590-b46d-2c5dc841a9a4

**Comment by thmsobrmlr** [comment]
I could reproduce it with that as well, thanks!

**Comment by thmsobrmlr** [comment]
Btw check you inbox, I've contacted you via your website for a small reward for your recent contributions. @Arun-chaitanya

**Comment by Arun-chaitanya** [comment]
@thmsobrmlr, Sorry but my emailjs credits got over, so did not get your mail. I have recharged my credits now, so I can get your mails now or here is my email arunchaitanya.iitkgp@gmail.com. You can drop a mail over here directly.

**Comment by thmsobrmlr** [comment]
> @thmsobrmlr, Sorry but my emailjs credits got over, so did not get your mail. I have recharged my credits now, so I can get your mails now or here is my email [arunchaitanya.iitkgp@gmail.com](mailto:arunchaitanya.iitkgp@gmail.com). You can drop a mail over here directly.

Just mailed your Gmail.

**Pull request merged**